### PR TITLE
Add Elastic Net algorithm

### PR DESCRIPTION
**Pull request by agorshk**
Add Elastic Net algorithm

### CODE REVIEW

**Comment by agorshk** [comment]
make test

**Comment by ShvetsKS** [comment]
can we consider it like cherry-pick of commit: 5c4178b12623f763e9dcc10a9e887a33b57f060d?
which was already reviewed?

File: algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+            result |= daal::services::internal::daal_memcpy_s(yTrainPtr, nDependentVariables * nRows * sizeof(algorithmFPType), yPtr,
+                                                              nDependentVariables * nRows * sizeof(algorithmFPType));
+            DAAL_CHECK(!result, services::ErrorMemoryCopyFailedInternal);
+        }
+
+        daal::internal::WriteRows<algorithmFPType, cpu> xBD(xTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(xBD);
+        daal::internal::WriteRows<algorithmFPType, cpu> yBD(yTrain.get(), 0, nRows);
+        DAAL_CHECK_BLOCK_STATUS(yBD);
+        algorithmFPType * xPtr = xBD.get();
```
Comment (ShvetsKS) [inline algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i]: algorithmFPType * const

and similar bellow

Comment (agorshk) [reply algorithms/kernel/elastic_net/elastic_net_train_dense_default_impl.i]: it was fixed, yes it's like a cherry-pick of commit: 5c4178b

**Comment by agorshk** [comment]
make test

**Comment by agorshk** [comment]
CI Id_ea499b3d-8c88-f139-a1f3-8cdcd4b723a1

**Pull request merged**